### PR TITLE
Add docs to CookiePolicy

### PR DESCRIPTION
**Pull request by pranavkm**
Contributes to https://github.com/dotnet/aspnetcore/issues/26397

### CODE REVIEW

File: src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs [hunk 1]
```diff
@@ -12,27 +12,49 @@
...
 
+        /// <summary>
+        /// Gets or sets the <see cref="CookiePolicyOptions"/>.
+        /// </summary>
         public CookiePolicyOptions Options { get; set; }
 
+        /// <summary>
+        /// Invokes the middleware.
+        /// </summary>
+        /// <param name="context"></param>
```
Comment (JamesNK) [inline src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs:56]: Should there be something here?

**Review by JamesNK — APPROVED** [review]

File: src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs (hunk 1 above)
Comment (pranavkm) [reply src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs:56]:
```suggestion
        /// <param name="context">The <see cref="HttpContext" />.</param>
```

File: src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs [hunk 2]
```diff
@@ -12,27 +12,49 @@
...
+        /// <returns></returns>
```
Comment (pranavkm) [inline src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs:57]:
```suggestion
```

**Comment by pranavkm** [comment]
@ajaybhargavb whenever you have a sec

File: src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs [hunk 3]
```diff
@@ -12,27 +12,48 @@
 
 namespace Microsoft.AspNetCore.CookiePolicy
 {
+    /// <summary>
+    /// Initializes a new instance of <see cref="CookiePolicyMiddleware"/>.
```
Comment (Tratcher) [inline src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs:16]: `A middleware that enforces policies for response cookies.`

File: src/Security/CookiePolicy/src/HttpOnlyPolicy.cs [hunk 4]
```diff
@@ -3,9 +3,21 @@
...
 namespace Microsoft.AspNetCore.CookiePolicy
 {
+    /// <summary>
+    /// Describes the HttpOnly behavior for cookies.
+    /// </summary>
     public enum HttpOnlyPolicy
     {
+        /// <summary>
+        /// The cookie does not have a configured HttpOnly behavior. This cookie can be accessed by
+        /// JavaScript <c>document.cookie</c> API.
```
Comment (Tratcher) [inline src/Security/CookiePolicy/src/HttpOnlyPolicy.cs:13]: `No policy is applied, response cookies are allowed to be sent with or without the HttpOnly attribute.`

**Pull request merged**